src/tests/ftest/util/yaml_utils.py

```diff
@@ -0,0 +1,485 @@
+             "hosts - the optional '<filter>' can be used to limit auto-detected "
+             "NVMe addresses, e.g. 'auto:Optane' for Intel Optane NVMe devices.  "
+             "To limit the device detection to either VMD controller or NVMe "
+             "devices the 'auto_vmd[:filter]' or 'auto_nvme[:<filter>]' keywords "
+             "can be used, respectively.  When using 'filter' with VMD controllers, "
+             "the filter is applied to devices managed by the controller, therefore "
+             "only selecting controllers that manage the matching devices.")
+    parser.add_argument(
+        "-o",
+        "--override",
+        action="store_true",
+        help="allow test yaml lists and NodeSets to be reduced or increased based upon "
+             "the user provided replacement values.  A separate --test_servers and "
+             "--test_clients entry must be specified when using this option.")
+    parser.add_argument(
+        "-tc", "--test_clients",
+        action="store",
+        help="comma-separated list of hosts to use as replacement values for "
+             "client placeholders in each test's yaml file")
+    parser.add_argument(
+        "-tm", "--timeout_multiplier",
+        action="store",
+        default=None,
+        type=float,
+        help="a multiplier to apply to each timeout value found in the test yaml")
+    parser.add_argument(
+        "-ts", "--test_servers",
+        action="store",
+        help="comma-separated list of hosts to use as replacement values for "
+             "server placeholders in each test's yaml file.  If the "
+             "'--test_clients' argument is not specified, this list of hosts "
+             "will also be used to replace client placeholders.")
+    parser.add_argument(
+        "-y", "--yaml_directory",
+        action="store",
+        default=None,
+        help="directory in which to write the modified yaml files. A temporary "
+             "directory - which only exists for the duration of the launch.py "
+             "command - is used by default.")
+
+
+def add_arguments_verbose(parser):
+    """Add yaml modification parameters to the provided command line parser.
+
+    Args:
+        parser (ArgumentParser): command line argument parser
+    """
+    parser.add_argument(
+        "-v", "--verbose",
+        action="count",
+        default=0,
+        help="verbosity output level. Specify multiple times (e.g. -vv) for "
+             "additional output")
+
+
+def main():
+    """Modify yaml files."""
+    log = get_console_logger()
+    log.info("--- YAML UTILS ---")
+
+    parser = ArgumentParser(prog="yaml_utils.py", description="Utilities used to modify yaml files")
+    add_arguments(parser)
+    add_arguments_verbose(parser)
+    parser.add_argument("files", nargs="*", type=str, help="yaml files to modify")
+    args = parser.parse_args()
+    log.debug("Arguments: {}".format(args))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions